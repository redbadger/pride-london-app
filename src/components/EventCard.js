// @flow
import React from "react";
import { View, StyleSheet, ImageBackground } from "react-native";
import formatDate from "date-fns/format";
import { lightNavyBlueColor } from "../constants/colors";
import Text from "./Text";
import SaveEventButton from "./SaveEventButton";
import { formattedEventPrice } from "../data/formatters";

type Props = {
  name: string,
  locationName: string,
  eventPriceLow: number,
  eventPriceHigh: number,
  startTime: string,
  endTime: string,
  imageUrl: string,
  isFree: boolean,
  isSaved: boolean,
  toggleSaved: boolean => void
};
const removeTimezoneFromDateString = isoString => isoString.slice(0, -6);

class EventCard extends React.PureComponent<Props> {
  static defaultProps = {
    isSaved: false
  };

  render() {
    const {
      name,
      locationName,
      startTime,
      endTime,
      imageUrl,
      eventPriceLow,
      eventPriceHigh,
      isFree,
      isSaved,
      toggleSaved
    } = this.props;
    const eventStartTime = removeTimezoneFromDateString(startTime);
    const eventEndTime = removeTimezoneFromDateString(endTime);
    const timeFormat = "HH:mm";
    const timeDisplay = `${formatDate(
      eventStartTime,
      timeFormat
    )} - ${formatDate(eventEndTime, timeFormat)}`;

    return (
      <View style={styles.eventCard}>
        <ImageBackground
          style={styles.imageContainer}
          source={{ uri: imageUrl }}
          resizeMode="cover"
        >
          <View style={styles.eventPriceContainer}>
            <Text type="price" color="whiteColor">
              {formattedEventPrice(isFree, eventPriceLow, eventPriceHigh)}
            </Text>
          </View>
        </ImageBackground>
        <View style={styles.eventCardDetails}>
          <View style={styles.column}>
            <View style={styles.row}>
              <View style={styles.column}>
                <Text
                  type="small"
                  color="lightNavyBlueColor"
                  style={styles.eventTime}
                >
                  {timeDisplay}
                </Text>
                <Text
                  numberOfLines={2}
                  type="h3"
                  color="lightNavyBlueColor"
                  style={styles.eventName}
                >
                  {name}
                </Text>
              </View>
              <View>
                <SaveEventButton active={isSaved} onPress={toggleSaved} />
              </View>
            </View>
            <Text type="small" color="lightNavyBlueColor">
              {locationName}
            </Text>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  row: {
    flex: 0,
    flexDirection: "row"
  },
  column: {
    flex: 0,
    flexDirection: "column",
    flexGrow: 1,
    flexShrink: 1
  },
  eventCard: {
    height: 108,
    flexDirection: "row",
    overflow: "hidden",
    borderRadius: 5
  },
  imageContainer: {
    width: 114,
    height: 108
  },
  eventPriceContainer: {
    height: 23,
    backgroundColor: lightNavyBlueColor,
    position: "absolute",
    paddingHorizontal: 5,
    justifyContent: "center"
  },
  eventCardDetails: {
    flex: 1,
    paddingLeft: 8
  },
  eventTime: {
    paddingTop: 12
  },
  eventName: {
    paddingTop: 4
  }
});

export default EventCard;
