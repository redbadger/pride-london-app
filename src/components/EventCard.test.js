// @flow
import React from "react";
import { shallow } from "enzyme";
import EventCard from "./EventCard";

it("renders correctly when isFree is true", () => {
  const output = shallow(
    <EventCard
      name="name"
      locationName="location"
      startTime="2018-09-16T12:12:12+01:00"
      endTime="2018-09-16T13:12:12+01:00"
      isFree
      eventPriceLow={0}
      eventPriceHigh={0}
      imageUrl="https://image.jpg"
      toggleSaved={() => {}}
    />
  );
  expect(output).toMatchSnapshot();
});
it("renders correctly when isFree is false and there is a price range", () => {
  const output = shallow(
    <EventCard
      name="name"
      locationName="location"
      startTime="2018-09-16T12:12:12+01:00"
      endTime="2018-09-16T13:12:12+01:00"
      isFree={false}
      eventPriceLow={12}
      eventPriceHigh={14}
      imageUrl="https://image.jpg"
      toggleSaved={() => {}}
    />
  );
  expect(output).toMatchSnapshot();
});
it("renders correctly when isFree is false and there's only one price", () => {
  const output = shallow(
    <EventCard
      name="name"
      locationName="location"
      startTime="2018-09-16T12:12:12+01:00"
      endTime="2018-09-16T13:12:12+01:00"
      isFree={false}
      eventPriceLow={12}
      eventPriceHigh={12}
      imageUrl="https://image.jpg"
      toggleSaved={() => {}}
    />
  );
  expect(output).toMatchSnapshot();
});
