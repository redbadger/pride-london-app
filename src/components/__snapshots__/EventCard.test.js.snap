// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when isFree is false and there is a price range 1`] = `
<Component
  style={
    Object {
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 108,
      "overflow": "hidden",
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.jpg",
      }
    }
    style={
      Object {
        "height": 108,
        "width": 114,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#2d2f7f",
          "height": 23,
          "justifyContent": "center",
          "paddingHorizontal": 5,
          "position": "absolute",
        }
      }
    >
      <Text
        color="whiteColor"
        markdown={false}
        markdownStyle={Object {}}
        type="price"
      >
        From £12
      </Text>
    </Component>
  </ImageBackground>
  <Component
    style={
      Object {
        "flex": 1,
        "paddingLeft": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 0,
            "flexDirection": "row",
          }
        }
      >
        <Component
          style={
            Object {
              "flex": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
        >
          <Text
            color="lightNavyBlueColor"
            markdown={false}
            markdownStyle={Object {}}
            style={
              Object {
                "paddingTop": 12,
              }
            }
            type="small"
          >
            12:12 - 13:12
          </Text>
          <Text
            color="lightNavyBlueColor"
            markdown={false}
            markdownStyle={Object {}}
            numberOfLines={2}
            style={
              Object {
                "paddingTop": 4,
              }
            }
            type="h3"
          >
            name
          </Text>
        </Component>
        <Component>
          <SaveEventButton
            active={false}
            onDark={false}
            onPress={[Function]}
          />
        </Component>
      </Component>
      <Text
        color="lightNavyBlueColor"
        markdown={false}
        markdownStyle={Object {}}
        type="small"
      >
        location
      </Text>
    </Component>
  </Component>
</Component>
`;

exports[`renders correctly when isFree is false and there's only one price 1`] = `
<Component
  style={
    Object {
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 108,
      "overflow": "hidden",
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.jpg",
      }
    }
    style={
      Object {
        "height": 108,
        "width": 114,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#2d2f7f",
          "height": 23,
          "justifyContent": "center",
          "paddingHorizontal": 5,
          "position": "absolute",
        }
      }
    >
      <Text
        color="whiteColor"
        markdown={false}
        markdownStyle={Object {}}
        type="price"
      >
        £12
      </Text>
    </Component>
  </ImageBackground>
  <Component
    style={
      Object {
        "flex": 1,
        "paddingLeft": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 0,
            "flexDirection": "row",
          }
        }
      >
        <Component
          style={
            Object {
              "flex": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
        >
          <Text
            color="lightNavyBlueColor"
            markdown={false}
            markdownStyle={Object {}}
            style={
              Object {
                "paddingTop": 12,
              }
            }
            type="small"
          >
            12:12 - 13:12
          </Text>
          <Text
            color="lightNavyBlueColor"
            markdown={false}
            markdownStyle={Object {}}
            numberOfLines={2}
            style={
              Object {
                "paddingTop": 4,
              }
            }
            type="h3"
          >
            name
          </Text>
        </Component>
        <Component>
          <SaveEventButton
            active={false}
            onDark={false}
            onPress={[Function]}
          />
        </Component>
      </Component>
      <Text
        color="lightNavyBlueColor"
        markdown={false}
        markdownStyle={Object {}}
        type="small"
      >
        location
      </Text>
    </Component>
  </Component>
</Component>
`;

exports[`renders correctly when isFree is true 1`] = `
<Component
  style={
    Object {
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 108,
      "overflow": "hidden",
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.jpg",
      }
    }
    style={
      Object {
        "height": 108,
        "width": 114,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#2d2f7f",
          "height": 23,
          "justifyContent": "center",
          "paddingHorizontal": 5,
          "position": "absolute",
        }
      }
    >
      <Text
        color="whiteColor"
        markdown={false}
        markdownStyle={Object {}}
        type="price"
      >
        Free
      </Text>
    </Component>
  </ImageBackground>
  <Component
    style={
      Object {
        "flex": 1,
        "paddingLeft": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 0,
            "flexDirection": "row",
          }
        }
      >
        <Component
          style={
            Object {
              "flex": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
        >
          <Text
            color="lightNavyBlueColor"
            markdown={false}
            markdownStyle={Object {}}
            style={
              Object {
                "paddingTop": 12,
              }
            }
            type="small"
          >
            12:12 - 13:12
          </Text>
          <Text
            color="lightNavyBlueColor"
            markdown={false}
            markdownStyle={Object {}}
            numberOfLines={2}
            style={
              Object {
                "paddingTop": 4,
              }
            }
            type="h3"
          >
            name
          </Text>
        </Component>
        <Component>
          <SaveEventButton
            active={false}
            onDark={false}
            onPress={[Function]}
          />
        </Component>
      </Component>
      <Text
        color="lightNavyBlueColor"
        markdown={false}
        markdownStyle={Object {}}
        type="small"
      >
        location
      </Text>
    </Component>
  </Component>
</Component>
`;
